### PR TITLE
Misc 6

### PR DESCRIPTION
**Pull request by jschueller**
- fix Pagmo MINLP input variables reordering
- add Function.setStopCallback method: the use-case is the interruption of Salome/YACS async eval API in Persalys
- Fix viewer z-order (again), else points appear behind grid
- #2596 
- Use kolmogorov distribution from scipy https://github.com/openturns/openturns/issues/2557: translate some code from Python and also include some of the needed C code
- Disable HMat by default (because of license)
- #2621
- #2602
- #2545 
- #2627 
- #2628 
- #2517
- Drop Tensor/Matrix.clean method call from str methods & tests
- Add nanoflann (from #2566)
- Add Matrix.frobeniusNorm (from #2566)

### CODE REVIEW

**Comment by sofianehaddad** [comment]
Maybe specific `kolmogorov` files (including `dd_*`) might be in a specific folder ? (

**Comment by jschueller** [comment]
> Maybe specific `kolmogorov` files (including `dd_*`) might be in a specific folder ? (

good idea, lets move it into /cephes

File: lib/src/Uncertainty/Distribution/TruncatedDistribution.cxx [hunk 1]
```diff
@@ -173,25 +174,37 @@ TruncatedDistribution * TruncatedDistribution::clone() const
...
   Distribution localDistribution(distribution_);
   String kind(localDistribution.getImplementation()->getClassName());
   // Delve into the antecedents until we get something which is not a truncated distribution
+  UnsignedInteger level = 1;
   while (kind == "TruncatedDistribution")
   {
     const TruncatedDistribution * truncatedDistribution = dynamic_cast<const TruncatedDistribution *>(localDistribution.getImplementation().get());
     localDistribution = truncatedDistribution->getDistribution();
     kind = localDistribution.getImplementation()->getClassName();
+    ++ level;
```
Comment (regislebrun) [inline lib/src/Uncertainty/Distribution/TruncatedDistribution.cxx:200]: Maybe you could add a comment on the fact that the nested truncation intervals have already been intersected during the nested range computations.

File: lib/src/Uncertainty/Distribution/cephes/dd_real.h [hunk 2]
```diff
@@ -0,0 +1,143 @@
+/*
```
Comment (regislebrun) [inline lib/src/Uncertainty/Distribution/cephes/dd_real.h:1]: So with this, we embed an extended precision library in addition to the optional mpfr/mpc multiprecision libraries used in SpecFunc. At one point we should coordinate this, in particular we could make it more visible troughout the whole library eg. to implement roundoff sensitive algorithms (orthonormalization, dot products...)

**Review by regislebrun — COMMENTED** [review]
Only minor remarks. Approved if it does not change by 100+ lines of codes since this review.

File: lib/src/Uncertainty/Distribution/cephes/dd_real.h (hunk 2 above)
Comment (jschueller) [reply lib/src/Uncertainty/Distribution/cephes/dd_real.h:1]: yes, I did not want to rewrite that part

File: lib/src/Base/Func/SpecFunc/openturns/SpecFunc.hxx [hunk 3]
```diff
@@ -31,6 +31,7 @@ BEGIN_NAMESPACE_OPENTURNS
 namespace SpecFunc
 {
 
+#ifndef SWIG
```
[the remark this answers is not in the document]
Comment (jschueller) [reply lib/src/Base/Func/SpecFunc/openturns/SpecFunc.hxx:34]: ok I can drop it but imho they are not necessary

**Pull request merged**